database/release_dao.py

```diff
@@ -0,0 +1,142 @@
+import sqlalchemy as sa
+from config import DATABASE_URL
+from models import Release
+from sqlalchemy.orm import sessionmaker
+
+# Needs to implement all CRUD operations on the given database.
+
+engine = sa.create_engine(DATABASE_URL)
+
+Session = sessionmaker(bind=engine)
+
+s = Session()
+
+
+def manCreateRelease(id, name, version, result):
+    """ Given int ID, string name, string version, and string result, 
+    creates a Release object, and inserts it into the database controlled
+    by the currently active session. 
+    Assumes that the id given is unique. """
+
+    try:
+        if id in getkeys():
+            raise Exception(
+                "That keyvalue (" + str(id) + ") is already in the database. "
+            )
+        currentRelease = Release(id=id, name=name, version=version, result=result)
+    except Exception as e:
+        print(e)
+    else:
+        s.add(currentRelease)
+
+
+def createRelease(name, version, result):
+    """ Given string name, string version, and string result, 
+    creates a Release object, and inserts it into the database controlled
+    by the currently active session. 
+    Autonatically generates an ID that will work based on the IDs already in the table. 
+    Uses the minimum unused integer (min 0). 
+    Depends on getkeys. """
+
+    currIDs = getkeys()
+    currIDs.sort()
+    minID = currIDs[0]
+    for id in currIDs[1:]:
+        if id != minID + 1:
+            minID += 1
+            break
+        else:
+            minID += 1
+    if minID == currIDs[-1]:
+        minID += 1
+
+    currentRelease = Release(id=minID, name=name, version=version, result=result)
+
+    s.add(currentRelease)
+
+
+def readRelease(id):
+    """ Given the (int) ID of the Release to be read, returns a string in the format:
+    'ID: %ID, Name: %name, Version: %version, Result: %result', where 
+    each %value is the value corresponding to the given ID. 
+    Assumes that the given ID is present in the database. """
+
+    outstring = "The ID (" + str(id) + ") is not in the database. "
+
+    try:
+        rel = s.query(Release).get(id)
+        assert rel != None
+    except Exception as e:
+        pass

```

Comment:
This `pass` is just suppressing the exception, we should actually raise the exception to let it bubble up to the caller function and let that other script decide what to do with it (if the Fast API router function tries to read a release ID and something goes wrong, we need to return a message to our user).